src/irmin-pack/irmin_pack.ml

```diff
@@ -308,7 +312,7 @@ struct
     end
 
     module Node = struct
-      module CA = Inode.Make (Config) (Pack) (H) (Node)
+      module CA = Inode.Make (Config) (H) (Pack) (Node)
```

Comment:
I'm a bit sad this is needed

Reply:
To me, this ordering is the sensible one since `Pack` relies on a hash implementation. I can avoid this with some extra shenanigans around exposing `index` types; is there a particular reason you wanted it this way round?

Reply:
no strong reason, I just didn't expect the index change to reflect in that file. But if it's simpler, that's fine.

Reply:
We could also avoid the ordering constraint by having `Inode` take a functor `H -> Pack` instead.

Reply:
let's avoid using higher-order functors :p